src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/OffenderServiceTest.kt

```diff
@@ -407,6 +419,91 @@ class OffenderServiceTest {
     )
   }
 
+  @Test
+  fun `getPrisonCaseNotesByNomsNumber returns NotFound when Case Notes request returns a 404`() {
+    val nomsNumber = "NOMS456"
+
+    every {
+      mockPrisonsApiClient.getCaseNotesPage(
+        nomsNumber = nomsNumber,
+        from = LocalDate.now().minusDays(prisonCaseNotesConfigBindingModel.lookbackDays!!.toLong()),
+        page = 0
+      )
+    } returns ClientResult.StatusCodeFailure(HttpMethod.GET, "/api/offenders/$nomsNumber/v2", HttpStatus.NOT_FOUND, null)
+
+    assertThat(offenderService.getPrisonCaseNotesByNomsNumber(nomsNumber) is AuthorisableActionResult.NotFound).isTrue
+  }
+
+  @Test
+  fun `getPrisonCaseNotesByNomsNumber returns Unauthorised when Case Notes request returns a 403`() {
+    val nomsNumber = "NOMS456"
+
+    every {
+      mockPrisonsApiClient.getCaseNotesPage(
+        nomsNumber = nomsNumber,
+        from = LocalDate.now().minusDays(prisonCaseNotesConfigBindingModel.lookbackDays!!.toLong()),
+        page = 0
+      )
+    } returns ClientResult.StatusCodeFailure(HttpMethod.GET, "/api/offenders/$nomsNumber/v2", HttpStatus.FORBIDDEN, null)
+
+    assertThat(offenderService.getPrisonCaseNotesByNomsNumber(nomsNumber) is AuthorisableActionResult.Unauthorised).isTrue
+  }
+
+  @Test
+  fun `getPrisonCaseNotesByNomsNumber returns Success, traverses pages from Client & excludes categories + subcategories`() {
+    val nomsNumber = "NOMS456"
+
+    val caseNotesPageOne = listOf(
+      CaseNoteFactory().produce(),
+      CaseNoteFactory().produce(),
+      CaseNoteFactory().withType("EXCLUDED_TYPE").produce()
+    )
+
+    val caseNotesPageTwo = listOf(
+      CaseNoteFactory().produce(),
+      CaseNoteFactory().produce(),
+      CaseNoteFactory().withType("TYPE").withSubType("EXCLUDED_SUBTYPE").produce()
+    )
+
+    every {
+      mockPrisonsApiClient.getCaseNotesPage(
+        nomsNumber = nomsNumber,
+        from = LocalDate.now().minusDays(prisonCaseNotesConfigBindingModel.lookbackDays!!.toLong()),
+        page = 0
+      )
+    } returns ClientResult.Success(
+      HttpStatus.OK,
+      CaseNotesPage(
+        totalElements = 6,
+        totalPages = 2,
+        number = 1,
+        content = caseNotesPageOne
+      )
+    )
+
+    every {
+      mockPrisonsApiClient.getCaseNotesPage(
+        nomsNumber = nomsNumber,
+        from = LocalDate.now().minusDays(prisonCaseNotesConfigBindingModel.lookbackDays!!.toLong()),
+        page = 1
+      )
+    } returns ClientResult.Success(
+      HttpStatus.OK,
+      CaseNotesPage(
+        totalElements = 4,
+        totalPages = 2,
+        number = 1,
```

Comment:
is `number` always going to be `1`, until our API implements implements its own pagination?

As far as I understand, the consumer supplies only a noms number, and our API traverses all the PrisonAPI pages necessary to collect all of the case notes for the configured time period, and our API returns all the case notes in a single response. If so should we make it clearer in our tests what `number` is and how it behaves?

I wonder if -- in order to retrieve the contents of multiple pages from the Prison API in a timely fashion -- we'll need to experiment with different page sizes? i.e. we ought to add a third config element to `prison-case-notes` in `application.yml`?

Reply:
This should have been 2 rather than 1 (have updated.)

But yes you are correct, our endpoint just takes noms number - then we follow the pages on Prison API until we've got all of the data.

Good shout on making the page size configurable - I've added that too